src/api/BundlePush/BundlePusher.ts

```diff
@@ -644,24 +644,58 @@ export class BundlePusher {
   private async gatherNodejsDiagnosticsFromCics(cicsSession: AbstractSession): Promise<boolean> {
     // Issue a CMCI get to the target CICSplex
     try {
-      const data: IResourceParms = { name: "CICSNodejsapp",
-                                     criteria: "BUNDLE=" + this.params.arguments.name,
-                                     regionName: this.params.arguments.scope,
-                                     cicsPlex: this.params.arguments.cicsplex };
-      const cmciResponse = await getResource(cicsSession, data);
-      const outputRecord = cmciResponse.response.records.cicsnodejsapp;
-      if (outputRecord === undefined) {
-        throw new Error("CICSNodejsapp output record not found.");
+      // First process each Region in the Scope
+      this.updateStatus("Querying Regions in Scope over CMCI");
+      const regionData: IResourceParms = { name: "CICSRegion",
+                                           regionName: this.params.arguments.scope,
+                                           cicsPlex: this.params.arguments.cicsplex };
+      const cmciRegionResponse = await getResource(cicsSession, regionData);
+      if (cmciRegionResponse === undefined ||
+          cmciRegionResponse.response === undefined ||
+          cmciRegionResponse.response.records === undefined ||
+          cmciRegionResponse.response.records.cicsregion === undefined) {
+        throw new Error("CICSRegion CMCI output record not found.");
       }
+      const outputRegionRecords = cmciRegionResponse.response.records.cicsregion;
+      let msg = "CICS Regions in Scope '" + this.params.arguments.scope + "' of CICSplex '" + this.params.arguments.cicsplex + "':\n";
+      this.issueMessage(msg);
 
       // We may have an array of records if there was more than one NODEJSAPP in the bundle
```

Comment:
I think this comment should refer to regions

Reply:
I was thinking the region stuff might want separating entirely as that could be useful for any Bundle whether it has Nodejs is content or not. I figured I'd return to that in another update though as it involves a bit more restructuring.